gems/sorbet-runtime/lib/types/props/utils.rb

```diff
@@ -43,6 +43,12 @@ def self.need_nil_write_check?(prop_rules)
     need_nil_read_check?(prop_rules) || T::Props::Utils.required_prop?(prop_rules)
   end
 
+  # The prop_rules indicate whether we should typecheck the value for the prop/field.
+  def self.need_type_check?(prop_rules)
```

Comment:
A micro-optimization for an edge case that still might be worthwhile: `return false if prop_rules[:type_object] == T.untyped`

Reply:
I tried this and it breaks some tests related to `setter_validate`. Currently, if `need_type_check?` is false we don't run any validations (e.g. `need_nil_write_check` or `setter_validate`) and the tests were set up where the props were `T.untyped` but had a `setter_validate` set.

I think this behavior makes sense since when using `checked(:never)` the prop basically operates as `attr_writer` (in fact that's how the setter is defined) with the small exception of being able to use default/factory.

While I think the behavior makes sense, I think maybe holding off and making this change/optimization on a separate PR may be a better option.